ext/opentelemetry-ext-flask/src/opentelemetry/ext/flask/__init__.py

```diff
@@ -1,97 +1,124 @@
 # Note: This package is not named "flask" because of
 # https://github.com/PyCQA/pylint/issues/2648
 
-import logging
+from logging import getLogger
 
-from flask import request as flask_request
+import flask
 
 import opentelemetry.ext.wsgi as otel_wsgi
 from opentelemetry import propagators, trace
 from opentelemetry.ext.flask.version import __version__
+from opentelemetry.patcher import BasePatcher
 from opentelemetry.util import time_ns
 
-logger = logging.getLogger(__name__)
+logger = getLogger(__name__)
 
 _ENVIRON_STARTTIME_KEY = "opentelemetry-flask.starttime_key"
 _ENVIRON_SPAN_KEY = "opentelemetry-flask.span_key"
 _ENVIRON_ACTIVATION_KEY = "opentelemetry-flask.activation_key"
 
 
-def instrument_app(flask):
-    """Makes the passed-in Flask object traced by OpenTelemetry.
+class _PatchedFlask(flask.Flask):
```

Comment:
I prefer the previous approach over deriving a new class. You can patch `__init__` instead of overriding it. I have no strong reason though.